### PR TITLE
Refactoring count( )

### PR DESCRIPTION
**Pull request by yingyi666**
I implemented the StorageEngine interface in both storage and mem as you said, and all the unit tests passed.

### CODE REVIEW

**Comment by yingyi666** [comment]
But I think you still need to check my pr carefully, because I may have missed something.

File: mem.go [hunk 1]
```diff
@@ -75,6 +75,17 @@ func (e *memEngine) DropCollection(name string) error {
 	return nil
 }
 
+func (e *memEngine) Count(q *Query) (int, error) {
+	var num int
```
Comment (ostafen) [inline mem.go:79]: Just because it looks to me more clear, in both methods I would initialize num to 0 like so:
```golang
num := 0
```

**Review by ostafen — CHANGES_REQUESTED** [review]
This looks good! Thank you for your contribution. I outlined only a minor change which is more a stylistic choice

**Comment by yingyi666** [comment]
i have initialize num to 0

File: mem.go (hunk 1 above)
Comment (yingyi666) [reply mem.go]: yes , u r right, initializing num to 0 make it easier to read

**Pull request merged**